src/components/container/InboxItemList/InboxItemList.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useContext, useState } from 'react';
+import './InboxItemList.css';
+import MyDataHelps, { InboxItem, InboxItemQueryParameters, InboxItemSortColumn, InboxItemStatus, InboxItemType, InboxMessage, InboxResource, InboxSurvey, SortOrder } from '@careevolution/mydatahelps-js';
+import { InboxMessageListItem, InboxResourceImageAlignment, InboxResourceListItem, InboxSurveyListItem, InboxSurveyVariant, LoadingIndicator } from '../../presentational';
+import { useInitializeView } from '../../../helpers/Initialization';
+import inboxDataService from '../../../helpers/Inbox/inbox-data';
+import { InboxItemListCoordinatorContext } from '../InboxItemListCoordinator';
+import { viewInboxMessage } from '../../view/InboxMessageView';
+import { InboxItemListPreviewState, previewData } from './InboxItemList.previewData';
+
+export interface InboxItemListProps {
+    previewState?: InboxItemListPreviewState;
+    title?: string;
+    type?: InboxItemType | InboxItemType[];
+    status?: InboxItemStatus | InboxItemStatus[];
+    category?: string;
+    sortOrder?: SortOrder;
+    sortColumn?: InboxItemSortColumn;
+    endsAfter?: string;
+    endsBefore?: string;
+    limit?: number;
+    messageViewerUrl?: string;
+    emptyText?: string;
+    showTitleWhenEmpty?: boolean;
+    surveyVariant?: InboxSurveyVariant;
+    resourceImageAlignment?: InboxResourceImageAlignment;
+    hideLoadingIndicator?: boolean;
+    onItemsLoaded?: (items: InboxItem[]) => void;
+    syncOnChanges?: boolean;
+    innerRef?: React.Ref<HTMLDivElement>;
+}
+
+export default function (props: InboxItemListProps) {
+    const [loading, setLoading] = useState<boolean>(true);
+    const [items, setItems] = useState<InboxItem[]>();
+
+    const onItemsLoaded = (items: InboxItem[]): void => {
+        setItems(items);
+        if (props.onItemsLoaded) {
+            props.onItemsLoaded(items);
+        }
+    };
+
+    const loadData = (): void => {
+        setLoading(true);
+
+        if (props.previewState) {
+            onItemsLoaded(previewData[props.previewState].items);
+            setLoading(false);
+            return;
+        }
+
+        let query: InboxItemQueryParameters = {
+            type: props.type,
+            status: props.status,
+            category: props.category,
+            sortOrder: props.sortOrder,
+            sortColumn: props.sortColumn,
+            endsAfter: props.endsAfter,
+            endsBefore: props.endsBefore
+        };
+        inboxDataService.loadInboxItems(query).then(items => {
+            onItemsLoaded(items);
+            setLoading(false);
+        });
+    };
+
+    let inboxContext = useContext(InboxItemListCoordinatorContext);
+    if (inboxContext && props.syncOnChanges) {
+        inboxContext.registerChangeListener(loadData);
+    }
+
+    useInitializeView(loadData, [], [props.previewState]);
+
+    if (loading && !items) {
+        return <div className="mdhui-inbox-item-list" ref={props.innerRef}>
```

Comment:
There is no class definition for mdhui-inbox-item-list

Reply:
Yeah, it didn't need any customization at this time.  I always like to have classes assigned to tags, though, so they can be styled later or by users of this library when necessary.